src/states/joining_peer.rs

```diff
@@ -143,6 +147,11 @@ impl JoiningPeer {
             };
             let join_request = JoinRequest {
                 elders_version,
+                knowledge: self
+                    .section_key_info
+                    .as_ref()
+                    .map(|info| info.version())
+                    .unwrap_or(0),
```

Comment:
Question to reviewers: when we are newly joining node, we are basically requesting a proof that starts at the root. However, we still can't properly verify such proof because we have no trusted keys. This means the message will contain unnecessary long proof for no real benefit. Is it worth optimizing this? For example changing the `knowledge` field to `Option<u64>` and setting it to `None` for newly joining nodes? So the target section could then use shorter proof to save same space?

Reply:
Yes, Maybe a new node could have knowledge of valid keys. but right now they do not so probably better to have it optional.

Latter we could add a mean for them to know genesis, or maybe with each update we could add all the trusted key at that point, but for now, we should make it clear that we do not require proof.